timeline-chart/src/components/time-graph-row-element.ts

```diff
@@ -87,14 +87,12 @@ export class TimeGraphRowElement extends TimeGraphComponent {
                     }
                 }
 
-                if (this.displayObject.children.length === 0) {
-                    const fontName = TimeGraphRowElement.fontController.getFontName(this._options.color ? this._options.color : 0, this._options.height - 2);
-                    const text = new PIXI.BitmapText(labelText, { fontName: fontName });
-                    this.displayObject.addChild(text);
+                if (this.displayObject.children.length !== 0) {
+                    this.displayObject.removeChildAt(0);
                 }
 
+                this.displayObject.addChild(new PIXI.BitmapText(textStr, { fontName: fontName }));
```

Comment:
With this for every redraw of a label a new BitmapText is created and the previous is removed. I would prefer to just update the text and reuse the BitmapText object. It would be good to investigate why the canvas renderer has trouble with it and we can find a different workaround. If not we still can consider this approach later on.

Reply:
So by looking at the overall design, we remove all the drawings on the canvas (like rows, states, arrows) and then redraw it again whenever there is an update (like zoom-in/out, changing width). So in my opinion, it makes more sense to do the same for the labels as well. (i.e clear the label every time we update and redraw again).

Reply:
> With this for every redraw of a label a new BitmapText is created and the previous is removed. I would prefer to just update the text and reuse the BitmapText object. It would be good to investigate why the canvas renderer has trouble with it and we can find a different workaround. If not we still can consider this approach later on.

Yes, we are removing the label and re-creating it again. As you said, the update works fine in the case of WebGL but somehow behaves differently for canvas-renderer. I found a somewhat related issue on the pixi repo https://github.com/pixijs/pixi.js/issues/4085.

I haven't been able to find the exact root cause for such a sporadic behavior but are you ok with this change until we come up with something more concrete? @MatthewKhouzam

Reply:
At the very least, let's have a cache or something. Thoughts?

Reply:
Ankush and I discussed offline (online) and he will try to make a pixibitmap repo. Maybe the bitmap objects don't need to be re-created, but only replaced for the draw to be correct. Either way, this is another limitation of PIXI.

Reply:
We have done some in depth study of the bitmaptext.

The conclusion: bitmaptext caches the font in a static cache. Therefore we can new BitmapText all day long with minimal performance impact! :) :) :) Feel free to merge it!